### DISCUSSION TITLE
Bevy GUI

### DISCUSSION THREAD

**Discussion by gerlvndv** [bevyengine/bevy#9037] [Q&A] [answered]
I find some pluses in creating a complex desktop ui using bevy.

- I like bevy
- ecs is flexible and convenient
- a lot of information about the engine

What disadvantages will i face ?

**Comment by nicopap** [comment, marked as answer]
Disadvantages:

- `bevy_ui` is extremely primitive, doesn't have any widget beside "button". You'll have to define every widgets yourself
- `bevy_ui` doesn't have an input solution, you'll have to inovate/reinvete the wheel to manage inputs
- `bevy_ui` doesn't have state management, you'll have to design a really nice bridge between the ECS world and your UI declaration layer to make it work
- `bevy_ui` has poor scene support, so you'll have to write a bit of code to get hot reloading and file-defined 

Basically you are going to write a UI framework on top of bevy. If you don't want to write a UI framework, but rather want to write a desktop application, I'd recommend either using a pre-existing bevy ui framework (such as bevy_egui, kayak or belly) or not using bevy for your desktop UI.

Note that existing bevy ui frameworks are unstable and fairly untested, so you run the risk of hitting a snag fairly quickly.
